src/Microsoft.TemplateSearch.Common/BlobStorageTemplateInfo.cs

```diff
@@ -103,18 +116,54 @@ private BlobStorageTemplateInfo(string identity, string name, IEnumerable<string
         [JsonProperty]
         public string? ThirdPartyNotices { get; private set; }
 
-        [JsonIgnore]
-        public IReadOnlyDictionary<string, IBaselineInfo> BaselineInfo => _baselineInfo.ToDictionary(kvp => kvp.Key, kvp => (IBaselineInfo)kvp.Value);
+        [JsonProperty]
+        public IReadOnlyDictionary<string, IBaselineInfo> BaselineInfo { get; private set; } = new Dictionary<string, IBaselineInfo>();
 
         [JsonIgnore]
         [Obsolete("This property is deprecated")]
         bool ITemplateInfo.HasScriptRunningPostActions { get; set; }
 
+        [JsonProperty]
         public IReadOnlyDictionary<string, string> TagsCollection { get; private set; } = new Dictionary<string, string>();
 
         public static BlobStorageTemplateInfo FromJObject(JObject entry)
         {
-            BlobStorageTemplateInfo info = entry.ToObject<BlobStorageTemplateInfo>();
+            string identity = entry.ToString(nameof(Identity)) ?? throw new ArgumentException($"{nameof(entry)} doesn't have {nameof(Identity)} property.", nameof(entry));
```

Comment:
`ToString` is already custom method, but it may return null when property is missing.
In fact this check is duplicate, it will be checked again in constructor. I'll remove it here.